### PR TITLE
Add validation code

### PR DESCRIPTION
**Pull request by thomasyu888**
Code review

### CODE REVIEW

File: sage_processing/validateGENIE.py [hunk 1]
```diff
@@ -0,0 +1,447 @@
...
+    :returns:             Text with all the errors in the clinical file
+    """
+
+
+    first_header = ['Chromosome','Start_Position','Reference_Allele','Tumor_Seq_Allele2','Tumor_Sample_Barcode']
+    correct_column_headers = ['Chromosome','Start_Position','Reference_Allele','Tumor_Seq_Allele2','Tumor_Sample_Barcode','t_alt_count','t_depth']
+    optional_headers = ['t_ref_count','n_depth','n_ref_count','n_alt_count']
+    
+    mutationDF = pd.read_csv(filePath,sep="\t",comment="#")
+
```
Comment (thomasyu888) [inline sage_processing/validateGENIE.py]: Add reference check

File: sage_processing/validateGENIE.py [hunk 2]
```diff
@@ -0,0 +1,447 @@
...
+def validateMAF(filePath):
+    """
+    This function validates the clinical file to make sure it adhere to the clinical SOP.
+    
+    :params filePath:     Path to mutation file
+
+    :returns:             Text with all the errors in the clinical file
+    """
+
+
```
Comment (thomasyu888) [inline sage_processing/validateGENIE.py]: Add check for if Reference and Variant Allele is not null

File: sage_processing/validateGENIE.py [hunk 3]
```diff
@@ -0,0 +1,447 @@
...
+    first_header = ['Chromosome','Start_Position','Reference_Allele','Tumor_Seq_Allele2','Tumor_Sample_Barcode']
```
Comment (thomasyu888) [inline sage_processing/validateGENIE.py]: first column is case sensitive and must be: Hugo_Symbol|Chromosome|Tumor_Sample_Barcode

**Comment by thomasyu888** [comment]
Confirmed the CBio Importer is case-insensitive

File: sage_processing/validateGENIE.py [hunk 4]
```diff
@@ -0,0 +1,469 @@
...
+        clinicalSampleDF[patientId] = ["-".join(samp.split("-")[0:3]) for samp in clinicalSampleDF[sampleId]]
+    #CHECK: All samples must have associated patient data 
+    if not all(clinicalSampleDF[patientId].isin(clinicalDF[patientId])):
+        total_error = total_error + "Sample: All samples must have associated patient information\n"
+    #CHECK: All patients must have associated sample data 
+    if not all(clinicalDF[patientId].isin(clinicalSampleDF[patientId])):
+        total_error = total_error + "Sample: All patients must have associated sample information\n"
+
+    #CHECK: PRIMARY_RACE
+    warn, error = checkMapping(clinicalDF,"PRIMARY_RACE","NAACCR_RACE_CODE_PRIMARY",[1,2,3,4,5,6,7,8,10,11,12,13,14,15,16,17,20,21,22,25,26,27,28,30,31,32,88,96,97,98,99,""])
```
Comment (kkdang) [inline sage_processing/validateGENIE.py]: Use table mappings

**Pull request merged**